src/components/groceryList/GroceryListForm.tsx

```diff
@@ -0,0 +1,327 @@
+    setItems([]);
+    setAddingItem(false);
+    form.reset();
+    fetchItems();
+  }, [isOpen, groceryList?.id]);
+
+  const FormSchema = z.object({
+    item_name: z.string().min(2, {
+      message: "Item name must be at least 2 characters.",
+    }),
+    quantity: z.coerce.number().min(1, {
+      message: "Quantity must be at least 1.",
+    }),
+    notes: z.string(),
+  });
+
+  const { toast } = useToast();
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      item_name: "",
+      quantity: 0,
+      notes: "",
+    },
+  });
+
+  function onSubmit(data: any) {
+    if (!groceryList) {
+      return;
+    }
+    data = {
+      ...data,
+      requester_group_id: group_id,
+      grocery_list_id: groceryList?.id,
+    };
+    setAddItemLoading(true);
+    toast({
+      title: "You submitted the following values:",
+      description: (
+        <pre className="mt-2 w-[340px] rounded-md bg-slate-950 p-4">
+          <code className="text-white">{JSON.stringify(data, null, 2)}</code>
+        </pre>
+      ),
+    });
+    apiService.post("/api/grocery-lists/add-item", data).then(() => {
+      toast({
+        title: "Success",
+        description: "Item added successfully",
+        variant: "success",
+      });
+      setAddingItem(false);
+      fetchItems();
+      setAddItemLoading(false);
+    });
+  }
+
+  const toggleItem = (id: number) => {
+    apiService
+      .post(
+        `/api/grocery-lists/toggle-item?item_id=${id}&grocery_list_id=${groceryList?.id}`
+      )
+      .catch(() => {
+        toast({
+          title: "Error",
+          description: "Failed to update item, please refresh and try again.",
+          variant: "destructive",
+        });
+      });
+    setItems((prevItems) =>
+      prevItems.map((item) =>
+        item.id === id ? { ...item, completed: !item.completed } : item
+      )
+    );
+  };
+
+  const deleteItem = (id: number) => {
+    apiService
+      .delete(`/api/grocery-lists/delete-item?item_id=${id}`)
+      .then(() => {
+        fetchItems();
+        toast({
+          title: "Success",
+          description: "Item deleted successfully",
+          variant: "success",
+        });
+      })
+      .catch(() => {
+        toast({
+          title: "Error",
+          description: "Failed to delete item, please refresh and try again.",
+          variant: "destructive",
+        });
+      });
+  };
+
+  const deleteList = () => {
+    if (!groceryList) {
+      return;
+    }
+    onDelete(groceryList.id);
+  };
+
+  return (
+    <>
+      {isOpen && groceryList && (
+        <Dialog open={isOpen} onOpenChange={onClose}>
+          <DialogContent>
+            <DialogHeader>
+              <DialogTitle>{groceryList.name}</DialogTitle>
+              <p className="text-muted-foreground text-sm">
+                Grocery list created on{" "}
+                {new Date(groceryList.created_at).toLocaleString()}
```

Comment:
this doesnt account for the time being stored in UTC, replacing the string in the date ctor with `${groceryList.created_at.replace("T", " ")} UTC` seems to fix it